cmd/gpu_plugin/gpu_plugin.go

```diff
@@ -494,6 +515,8 @@ func main() {
 	flag.BoolVar(&opts.resourceManagement, "resource-manager", false, "fractional GPU resource management")
 	flag.IntVar(&opts.sharedDevNum, "shared-dev-num", 1, "number of containers sharing the same GPU device")
 	flag.StringVar(&opts.preferredAllocationPolicy, "allocation-policy", "none", "modes of allocating GPU devices: balanced, packed and none")
+	flag.StringVar(&opts.nfdResourceFileName, "nfd-resource-filename", "intel-gpu-resources.txt", "filename for writing labels under NFD's features.d directory")
+	flag.BoolVar(&opts.disableResourceLabelExport, "disable-resource-label-export", false, "export labels from node's GPUs and store them for NFD")
```

Comment:
I thought I would not comment on this PR much but changed my mind.

I'd propose this initial version uses `-resource-management` boolean to decide if the resource labels are exported. With that, we would not have to make CRD API changes and it would also keep the UX simpler (due to smaller amount of config options and their meanings). Would that be possible?

Reply:
The functionality would change slightly. An example label set:
```
gpu.intel.com/cards: card1.card2
gpu.intel.com/gpu-numbers: "1.2"
gpu.intel.com/graphics_version: "12.60"
gpu.intel.com/media_version: "12.60"
gpu.intel.com/numa-gpu-map: 0-1_1-2
gpu.intel.com/platform_XYZ.count: "2"
gpu.intel.com/platform_XYZ.present: "true"
gpu.intel.com/platform_XYZ.tiles: "2"
gpu.intel.com/platform_gen: "12"
```

From these, `numa-gpu-map`, `cards` and `gpu-numbers` are used by GAS (= -resource-management). The rest could be used for Pod's nodeSelector or nodeAffinity. I know some of them are used for our internal validation without GAS.

Reply:
One addition though. The labels with "platform_" prefix are only available IF nfd is running as root (=debugfs access) which it isn't by default.

Reply:
our own documentation says `media_version`, `graphics_version` and `platform_gen` are deprecated so it sounds like a good opportunity to enable `platform_XYZ` ones via `NodeFeatureRules` and have the rest enabled when `-resource-management` is enabled.